### PR TITLE
Fix near-blake2 dependency

### PR DESCRIPTION
**Pull request by mrLSD**
Issue related to `blake2_f` function compatibility.

Related to:
* [EIP-152](https://eips.ethereum.org/EIPS/eip-152)
* [Blake2 RFC-7693](https://datatracker.ietf.org/doc/html/rfc7693#section-3.2)

Realted to *EIP-152* `compress` function should accept parameters: `h, m, t, f`

### CODE REVIEW

**Review by joshuajbouw — CHANGES_REQUESTED** [review]
While this does work, do remember the discussion we had and the requirements that we should have aurora-blake2 to be that commit, then we use our own library for that.

**Comment by mrLSD** [comment]
At the moment after the research, it became obvious that the `compress` upstream `blake2` function does not meet the requirements for `EIP-152`.
What does it mean:
* We can't directly set parameters: `h, m, t, f`
* We can't set the `round` parameter
* Implementation looks like a "black box" - if we compare m with the reference implementation of `[Blake2 RFC-7693](https://datatracker.ietf.org/doc/html/rfc7693#section-3.2)`

`near-blake2` (exact commit) solve that issue.

**Comment by joshuajbouw** [comment]
As we had discussed yeah, it's quite a bit unfortunate. The RFC is actually intended to be quite open. It is always scary to have to alter a cryptographic function to fit a requirement :S. Would be good to point that out to the authors, and figure out a way forward.

File: engine-precompiles/Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ autobenches = false
 aurora-engine-types = { path = "../engine-types", default-features = false }
 aurora-engine-sdk = { path = "../engine-sdk", default-features = false }
 base64 = { version = "0.13.0", default-features = false, features = ["alloc"] }
-near-blake2 = { git = "https://github.com/near/near-blake2.git", version = "0.9.1", default-features = false }
+near-blake2 = { git = "https://github.com/near/near-blake2.git", rev = "cebb7df79608a7058017940830d37c37d55d21fe", default-features = false }
```
Comment (joshuajbouw) [inline engine-precompiles/Cargo.toml:19]: This needs to be changed to Aurora's

**Review by joshuajbouw — CHANGES_REQUESTED** [review]

**Comment by artob** [comment]
Task linked: [CU-290q9nc Engine v1.5.2](https://app.clickup.com/t/290q9nc)

**Comment by artob** [comment]
Task linked: [CU-290q9nc Engine v1.5.2](https://app.clickup.com/t/290q9nc)

**Review by joshuajbouw — APPROVED** [review]

**Pull request merged**